src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+			this._colorComponent = options.colorComponent;
+		}
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+	public set colorComponent(value: ColorComponent) {
+		if (!ColorComponent[value]) {
+			throw new Error(
+				"colorComponent must be a valid color component index.",
+			);
+		}
+		this._colorComponent = value;
+	}
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(1);
+		this.payload[0] = this._colorComponent;
+		return super.serialize();
+	}
+}
+
+export interface ColorSwitchCCSetOptions extends CCCommandOptions, ColorTable {
+	duration?: Duration;
+}
+
+@CCCommand(ColorSwitchCommand.Set)
+export class ColorSwitchCCSet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | ColorSwitchCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			// Populate properties from options object
+			this._colorTable = {
+				warmWhite: options.warmWhite,
+				coldWhite: options.coldWhite,
+				red: options.red,
+				green: options.green,
+				blue: options.blue,
+				cyan: options.cyan,
+				purple: options.purple,
+				index: options.index,
+			};
+			this.duration = options.duration ?? new Duration(1, "seconds");
+		}
+	}
+
+	private _colorTable: ColorTable;
+	public get colorTable(): ColorTable {
+		return this._colorTable;
+	}
+	public set colorTable(value: ColorTable) {
+		this._colorTable = {
+			warmWhite: value.warmWhite,
+			coldWhite: value.coldWhite,
+			red: value.red,
+			green: value.green,
+			blue: value.blue,
+			cyan: value.cyan,
+			purple: value.purple,
+			index: value.index,
+		};
+	}
+
+	private _duration: Duration = new Duration(0, "default");
+	public get duration(): Duration {
+		return this._duration;
+	}
+	public set duration(value: Duration) {
+		this._duration = value;
+	}
+
+	public serialize(): Buffer {
+		const populatedColorKeys = ColorKeys.filter(
+			(x) => !isNaN(this._colorTable[x] as any),
+		);
+		const populatedColorCount = populatedColorKeys.length;
+		this.payload = Buffer.allocUnsafe(
+			1 + populatedColorCount * 2 + (this.version >= 2 ? 1 : 0),
+		);
+		this.payload[0] = populatedColorCount & 0b11111;
+		let i = 1;
+		for (const key of populatedColorKeys) {
+			const value = this._colorTable[key];
+			const component = ColorComponentMap[key];
+			this.payload[i] = component;
+			this.payload[i + 1] = clamp(value!, 0, 0xff);
+			i += 2;
+		}
+		if (this.version >= 2) {
+			this.payload[i] = this._duration.serializeSet();
+		}
+		return super.serialize();
+	}
+}
+
+export interface ColorSwitchCCStartLevelChangeOptions extends CCCommandOptions {
+	colorComponent: ColorComponent;
+	down?: boolean;
```

Comment:
Please take a look at `MultilevelSwitchCCStartLevelChange` and implement this in a similar fashion.

Reply:
LevelChangeDirection could probably be moved to a shared file then (probably somewhere in `./values`)